lib/maths/CSeasonalTime.cc

```diff
@@ -198,21 +222,34 @@ CGeneralPeriodTime* CGeneralPeriodTime::clone() const {
     return new CGeneralPeriodTime(*this);
 }
 
-bool CGeneralPeriodTime::fromString(const std::string& value) {
+bool CGeneralPeriodTime::fromString(std::string value) {
+    std::size_t delimiter{value.find(DELIMITER)};
```

Comment:
`rfind` might be more robust to future changes given how the string is being concatenated in the persistence method.

Reply:
See [2e9d47c](https://github.com/elastic/ml-cpp/pull/821/commits/2e9d47cd0955a5a8c22509e8c239e50438b3c5e0)